src/update_agent/actor.rs

```diff
@@ -339,27 +340,23 @@ impl UpdateAgentInfo {
         trace!("trying to stage an update");
 
         let target = release.clone();
-        let deploy_outcome = self.attempt_deploy(target).await;
+        let deploy_outcome = self
+            .attempt_deploy(target)
+            .and_then(|release| self.confirm_valid_stream(state, release))
+            .await;
 
-        match deploy_outcome {
-            Ok(release) => {
-                // Sanity check that the deployment we just staged is from the correct stream.
-                self.check_stream(state, release.clone()).await;
-            }
-            Err(e) => {
-                log::error!("failed to stage deployment: {}", e);
-                let release_ver = release.version.clone();
-                let fail_count =
-                    UpdateAgentInfo::deploy_attempt_failed(release, &mut state.machine_state);
-                let msg = format!(
-                    "trying to stage {} ({} failed deployment attempt{})",
-                    release_ver,
-                    fail_count,
-                    if fail_count > 1 { "s" } else { "" }
-                );
-                utils::update_unit_status(&msg);
-                log::trace!("{}", msg);
-            }
+        if let Err(e) = deploy_outcome {
+            log::error!("failed to stage deployment: {}", e);
+            let fail_count =
+                UpdateAgentInfo::deploy_attempt_failed(&release, &mut state.machine_state);
+            let msg = format!(
+                "trying to stage {} ({} failed deployment attempt{})",
+                release.version,
+                fail_count,
+                if fail_count > 1 { "s" } else { "" }
```

Comment:
I know this existed before, but I try to avoid this by doing e.g.:
`trying to stage {} (failed attempts: {})` - it's OK to have "failed attempts: 1" for example.

Reply:
Yes, it'd be good to tweak it here, I'll bundle that in the next rebase.